apache/zeppelin
# [ZEPPELIN-2430] The xAxis text in the line and area chart overlaps

### What is this PR for?
Like in the #2133 case, the xAxis text of area chart and line chart is overlapping.
For reusing form, I created a html template file for how to show xLabel and, in `rotate` button checked, show up `input` box for set value of rotate degree (default value is -45).

### What type of PR is it?
[Bug Fix | Improvement]

### What is the Jira issue?
* [ZEPPELIN-2430 - The xlabel text in the line and area graph overlaps](https://issues.apache.org/jira/browse/ZEPPELIN-2430)

### How should this be tested?
1. Click `settings` button in the barChart 
2. Check to whether there are `default`, `rotate`, `hide` button or not.
3. Test to click each button and whether it works well or not for each situation.
4. In rotate button checked, check to set value and showing xAixs rotate degree properly.
5. In the areaChart and lineChart, test with above steps.

### Screenshots (if appropriate)
[Before]
![image](https://cloud.githubusercontent.com/assets/8110458/25982243/cffe5510-3715-11e7-8ca9-5c9f1147b61f.png)

![image](https://cloud.githubusercontent.com/assets/8110458/25982234/bef46cc8-3715-11e7-9839-ee10ebba3423.png)

[After]
![z_overlap_in_text](https://cloud.githubusercontent.com/assets/8110458/25982228/a9659aa8-3715-11e7-938c-8602f5b9b270.gif)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No
